src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionExecutor.cs

```diff
@@ -31,6 +33,20 @@ public FunctionDescriptor Function
         }
 
         public async Task<FunctionResult> TryExecuteAsync(TriggeredFunctionData input, CancellationToken cancellationToken)
+        {
+            if (_retryManagerProvider != null)
+            {
+                IRetryManager retryManager = _retryManagerProvider.Create(_descriptor.RetryAttribute);
```

Comment:
Ideally on the execution hotpath, we wouldn't have to create a new manager for each invocation. Instead, can't you create it once on construction (or get it from DI) and have the ExecuteWithRetries handle state internally (i.e. the manager instance itself is stateless)?

Reply:
Actually, why do we need an IRetryManagerProvider then at that point? It can just be a single IRetryManager interface, with the instance created only once.